streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBStoreTest.java

```diff
@@ -872,62 +879,65 @@ public void shouldPerformRangeQueriesWithCachingDisabled() {
         context.setTime(1L);
         store.put(1, "hi");
         store.put(2, "goodbye");
-        final KeyValueIterator<Integer, String> range = store.range(1, 2);
-        assertEquals("hi", range.next().value);
-        assertEquals("goodbye", range.next().value);
-        assertFalse(range.hasNext());
+        try (final KeyValueIterator<Integer, String> range = store.range(1, 2)) {
+            assertEquals("hi", range.next().value);
+            assertEquals("goodbye", range.next().value);
+            assertFalse(range.hasNext());
+        }
     }
 
     @Test
     public void shouldPerformAllQueriesWithCachingDisabled() {
         context.setTime(1L);
         store.put(1, "hi");
         store.put(2, "goodbye");
-        final KeyValueIterator<Integer, String> range = store.all();
-        assertEquals("hi", range.next().value);
-        assertEquals("goodbye", range.next().value);
-        assertFalse(range.hasNext());
+        try (final KeyValueIterator<Integer, String> range = store.all()) {
+            assertEquals("hi", range.next().value);
+            assertEquals("goodbye", range.next().value);
+            assertFalse(range.hasNext());
+        }
     }
 
     @Test
     public void shouldCloseOpenRangeIteratorsWhenStoreClosedAndThrowInvalidStateStoreOnHasNextAndNext() {
         context.setTime(1L);
         store.put(1, "hi");
         store.put(2, "goodbye");
-        final KeyValueIterator<Integer, String> iteratorOne = store.range(1, 5);
-        final KeyValueIterator<Integer, String> iteratorTwo = store.range(1, 4);
+        try (final KeyValueIterator<Integer, String> iteratorOne = store.range(1, 5);
+             final KeyValueIterator<Integer, String> iteratorTwo = store.range(1, 4)) {
 
-        assertTrue(iteratorOne.hasNext());
-        assertTrue(iteratorTwo.hasNext());
+            assertTrue(iteratorOne.hasNext());
+            assertTrue(iteratorTwo.hasNext());
 
-        store.close();
+            store.close();
 
-        try {
-            iteratorOne.hasNext();
-            fail("should have thrown InvalidStateStoreException on closed store");
-        } catch (final InvalidStateStoreException e) {
-            // ok
-        }
+            try {
+                iteratorOne.hasNext();
+                fail("should have thrown InvalidStateStoreException on closed store");
+            } catch (final InvalidStateStoreException e) {
```

Comment:
Might be nice to rewrite this using `assertThrows` ? (Similar below.)

Reply:
Good suggestion! Updated. Thank you.